scaper/core.py

```diff
@@ -2103,13 +2241,38 @@ def generate(self,
         soundscape_audio, event_audio_list = None, None
 
         # Generate the audio and save to disk
+        scale_factor = 1.0
+        ref_db_change = 0
         if not no_audio:
-            soundscape_audio, event_audio_list = \
-                self._generate_audio(audio_path, ann, reverb=reverb,
+            soundscape_audio, event_audio_list, scale_factor, ref_db_change = \
+                self._generate_audio(audio_path, ann,
+                                     reverb=reverb,
                                      save_isolated_events=save_isolated_events,
                                      isolated_events_path=isolated_events_path,
-                                     disable_sox_warnings=disable_sox_warnings)
-
+                                     disable_sox_warnings=disable_sox_warnings,
+                                     fix_clipping=fix_clipping,
+                                     peak_normalization=peak_normalization)
+
+        # TODO: Stick to heavy handed overwriting for now, in the future we
+        #  should consolidate this with what happens inside _instantiate().
+        ann.sandbox.scaper.audio_path = audio_path
+        ann.sandbox.scaper.jams_path = jams_path
+        ann.sandbox.scaper.allow_repeated_label = allow_repeated_label
+        ann.sandbox.scaper.allow_repeated_source = allow_repeated_source
+        ann.sandbox.scaper.reverb = reverb
+        ann.sandbox.scaper.fix_clipping = fix_clipping
+        ann.sandbox.scaper.peak_normalization = peak_normalization
+        ann.sandbox.scaper.save_isolated_events = save_isolated_events
+        ann.sandbox.scaper.isolated_events_path = isolated_events_path
+        ann.sandbox.scaper.disable_sox_warnings = disable_sox_warnings
+        ann.sandbox.scaper.no_audio = no_audio
+        ann.sandbox.scaper.txt_path = txt_path
+        ann.sandbox.scaper.txt_sep = txt_sep
+        ann.sandbox.scaper.disable_instantiation_warnings = disable_instantiation_warnings
+        ann.sandbox.scaper.peak_normalization_scale_factor = scale_factor
+        ann.sandbox.scaper.ref_db_change = ref_db_change
+        ann.sandbox.scaper.ref_db_generated = self.ref_db + ref_db_change
```

Comment:
Duplicates code from `generate_from_jams` (this code is a superset of that code). So might be worth refactoring, as I mentioned before.

Reply:
I've created #135 so we address this in a future PR.